### PR TITLE
Feature/Manage input rules

### PR DESCRIPTION
**Pull request by k-y0u**
Add attribute rules and required management, for detail panel.

### CODE REVIEW

File: src/composables/QuasarFieldRule.js [hunk 1]
```diff
@@ -58,3 +58,77 @@ export function isUniqueFileLabel(t, tree, value) {
...
+ * otherwise the translated error message.
+ */
+export function isRequired(t, value, required) {
+  return !required
+    || !(value === null || value === undefined || value === '')
+    || t('errors.rules.required');
+}
+
+/**
+ * Check if the value length is superior or equal to min.
```
Comment (justinehell) [inline src/composables/QuasarFieldRule.js:77]: `superior` -> `greater than` for consistency with translation

File: src/composables/QuasarFieldRule.js [hunk 2]
```diff
@@ -58,3 +58,77 @@ export function isUniqueFileLabel(t, tree, value) {
...
+ * @param {Boolean} min - Rules value.
+ * @return {Boolean|String} Return true if value is valid,
+ * otherwise the translated error message.
+ */
+export function isStringTooShort(t, value, min) {
+  return !min || value.length >= min || t('errors.rules.string.min', { min });
+}
+
+/**
+ * Check if the value length is inferior or equal to max.
```
Comment (justinehell) [inline src/composables/QuasarFieldRule.js:89]: `inferior` -> `less than` for consistency with translation

File: src/composables/QuasarFieldRule.js [hunk 3]
```diff
@@ -58,3 +58,77 @@ export function isUniqueFileLabel(t, tree, value) {
...
+ * @param {Boolean} max - Rules value.
+ * @return {Boolean|String} Return true if value is valid,
+ * otherwise the translated error message.
+ */
+export function isStringTooLong(t, value, max) {
+  return !max || value.length <= max || t('errors.rules.string.max', { max });
+}
+
+/**
+ * Check if the value match the RegExp.
```
Comment (justinehell) [inline src/composables/QuasarFieldRule.js:101]: `match` -> `matches`

File: src/composables/QuasarFieldRule.js [hunk 4]
```diff
@@ -58,3 +58,77 @@ export function isUniqueFileLabel(t, tree, value) {
...
+ * @param {Boolean} regexp - Rules value.
+ * @return {Boolean|String} Return true if value is valid,
+ * otherwise the translated error message.
+ */
+export function isStringMatchingRegExp(t, value, regexp) {
+  return !regexp || (new RegExp(regexp)).test(value) || t('errors.rules.string.regexp', { regexp });
+}
+
+/**
+ * Check if the value is superior or equal to min.
```
Comment (justinehell) [inline src/composables/QuasarFieldRule.js:113]: same as with `isStringTooShort`

File: src/composables/QuasarFieldRule.js [hunk 5]
```diff
@@ -58,3 +58,77 @@ export function isUniqueFileLabel(t, tree, value) {
...
+ * @param {Boolean} min - Rules value.
+ * @return {Boolean|String} Return true if value is valid,
+ * otherwise the translated error message.
+ */
+export function isNumberTooSmall(t, value, min) {
+  return !min || value >= min || t('errors.rules.number.min', { min });
+}
+
+/**
+ * Check if the value is inferior or equal to max.
```
Comment (justinehell) [inline src/composables/QuasarFieldRule.js:125]: same

File: src/components/inputs/SelectInput.vue [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <q-select
+    clearable
+    v-model="localValue"
+    :options="options"
+    :rules="[
+      (value) => isRequired($t, value, attribute.definition.required),
```
Comment (justinehell) [inline src/components/inputs/SelectInput.vue:7]: forgot to add the `definition?.required` ?

**Review by Zorin95670 — DISMISSED** [review]

File: changelog.md [hunk 7]
```diff
@@ -52,5 +52,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add an input in InputWrapper to allow updating the key (name) of the unreferenced attribute
 - Add create and delete action for unreferenced attribute
 - Add process.env variable for icomorphic-git cors proxy url
+- Add required and rules management.
```
Comment (justinehell) [inline changelog.md:55]: no dot

File: src/components/inputs/SelectInput.vue (hunk 6 above)
Comment (k-y0u) [reply src/components/inputs/SelectInput.vue:7]: on purpose, as we discuss yesterday.
no def = no selectInput

File: changelog.md (hunk 7 above)
Comment (k-y0u) [reply changelog.md:55]: Done ! Period !

File: src/composables/QuasarFieldRule.js (hunk 1 above)
Comment (k-y0u) [reply src/composables/QuasarFieldRule.js:77]: done

File: src/composables/QuasarFieldRule.js (hunk 3 above)
Comment (k-y0u) [reply src/composables/QuasarFieldRule.js:101]: done

File: src/composables/QuasarFieldRule.js (hunk 4 above)
Comment (k-y0u) [reply src/composables/QuasarFieldRule.js:113]: done

File: src/composables/QuasarFieldRule.js (hunk 5 above)
Comment (k-y0u) [reply src/composables/QuasarFieldRule.js:125]: done

**Review by justinehell — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**